core/src/main/java/org/elasticsearch/transport/local/LocalTransportChannel.java

```diff
@@ -102,16 +101,28 @@ public void sendResponse(Throwable error) throws IOException {
     }
 
     private void sendResponseData(byte[] data) {
-        sourceTransport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
-        reservedBytes = 0;
-        targetTransport.workers().execute(() -> {
-            ThreadContext threadContext = targetTransport.threadPool.getThreadContext();
-            try (ThreadContext.StoredContext ignore = threadContext.stashContext()){
-                targetTransport.messageReceived(data, action, sourceTransport, version, null);
-            }
-        });
+        if (closed.get()) {
```

Comment:
I think we can just call `close();` here and free the bytes before we call the execute on the targetTransport